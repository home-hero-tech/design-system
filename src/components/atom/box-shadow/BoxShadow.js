import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import css from './BoxShadow.module.scss';

const BoxShadow = ({ level, children }) => {
  const classes = classNames({
    [css['elevation--1']]: level === 1,
    [css['elevation--2']]: level === 2,
    [css['elevation--3']]: level === 3,
    [css['elevation--4']]: level === 4,
    [css['elevation--5']]: level === 5
  });

  return <div className={classes}>{children}</div>;
};

BoxShadow.propTypes = {
  level: PropTypes.oneOf([1, 2, 3, 4, 5]).isRequired,
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
};

BoxShadow.defaultProps = {
  children: null
};

export default BoxShadow;
