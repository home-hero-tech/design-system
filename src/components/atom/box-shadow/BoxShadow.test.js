import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BoxShadow from './BoxShadow';

describe('BoxShadow', () => {
  it('should match with the snapshot', () => {
    const children = <div>BoxShadow</div>;
    const wrapper = shallow(<BoxShadow level={1}>{children}</BoxShadow>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
