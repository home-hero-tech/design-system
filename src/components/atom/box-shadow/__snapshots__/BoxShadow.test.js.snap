// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxShadow should match with the snapshot 1`] = `
<div
  className="c-box-shadow--elevation-1 c-box-shadow--radius-2"
>
  <div>
    BoxShadow
  </div>
</div>
`;
